modules/ocf_hpc/manifests/compute.pp

```diff
@@ -0,0 +1,34 @@
+class ocf_hpc::compute {
+  require ocf_hpc
+
+  # install proprietary nvidia drivers
+  package { ['nvidia-driver', 'nvidia-settings', 'nvidia-cuda-toolkit']:; }
```

Comment:
cuDNN is a userspace library, so users can install that on their Singularity containers (nvidia apparently provides Docker images with cuDNN which can be built to Singularity). Also, there's no cuDNN package for Debian, so it would be a pain to puppet.